packages/components/src/utils/useExitingAnimation/useExitingAnimation.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react'
+
+/** a utility hook for handling an the addition and removal of an animation and exit animation class */
+export const useExitingAnimation = (
+  shouldAnimate: boolean | undefined,
+  animatingClass: string,
+  exitingClass: string,
+  exitDuration: number = 1000,
+): string | undefined => {
```

Comment:
```suggestion
): string => {
```

I don't think we need `undefined` here because `animationClass` is always a string

Reply:
This also potentially brings up a larger can of worms about the `@typescript-eslint/explicit-function-return-type` rule - I'm not a big fan of explicit function return types because it's easy to lie to the TS compiler about what a function could return. In this case, the `animationClass` variable in `Well.tsx` was typed as `string | undefined` and it gets unnecessarily messy if we need to keep adding in `undefined` guards

Reply:
Thanks for the spot 👓 I've removed the undefined since as you say, is always returned as a string 👍 As to the larger can of worms question around explicit returns and our ts rules, I'm not strong in favor for and against. I believe this rule was added to ensure we don't rub up against consumers that have stricter types - https://cultureamp.slack.com/archives/C02NUQ27G56/p1683244063168049?thread_ts=1683242725.889089&cid=C02NUQ27G56 - don't know if this is still relevant 🤷

Reply:
Yeah cool I don't think we need to take any action on the rule for now since this could just be a once-off thing - so we'll just keep it under observation 👀